apply-dictionary-renderer-to-feature-layer/src/main/java/com/esri/arcgismaps/sample/applydictionaryrenderertofeaturelayer/MainActivity.kt

```diff
@@ -0,0 +1,119 @@
+ *
+ */
+
+package com.esri.arcgismaps.sample.applydictionaryrenderertofeaturelayer
+
+import android.os.Bundle
+import android.util.Log
+import androidx.appcompat.app.AppCompatActivity
+import androidx.databinding.DataBindingUtil
+import androidx.lifecycle.lifecycleScope
+import com.arcgismaps.ApiKey
+import com.arcgismaps.ArcGISEnvironment
+import com.arcgismaps.data.Geodatabase
+import com.arcgismaps.mapping.ArcGISMap
+import com.arcgismaps.mapping.BasemapStyle
+import com.arcgismaps.mapping.Viewpoint
+import com.arcgismaps.mapping.layers.FeatureLayer
+import com.arcgismaps.mapping.symbology.DictionaryRenderer
+import com.arcgismaps.mapping.symbology.DictionarySymbolStyle
+import com.esri.arcgismaps.sample.applydictionaryrenderertofeaturelayer.databinding.ActivityMainBinding
+import com.google.android.material.snackbar.Snackbar
+import kotlinx.coroutines.launch
+import java.io.File
+
+class MainActivity : AppCompatActivity() {
+
+    private val TAG = MainActivity::class.java.simpleName
+
+    // set up data binding for the activity
+    private val activityMainBinding: ActivityMainBinding by lazy {
+        DataBindingUtil.setContentView(this, R.layout.activity_main)
+    }
+
+    private val mapView by lazy {
+        activityMainBinding.mapView
+    }
+
+    private val provisionPath: String by lazy {
+        getExternalFilesDir(null)?.path.toString() + File.separator + getString(R.string.app_name)
+    }
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+
+        // authentication with an API key or named user is
+        // required to access basemaps and other location services
+        ArcGISEnvironment.apiKey = ApiKey.create(BuildConfig.API_KEY)
+        lifecycle.addObserver(mapView)
+
+        // create and add a map with a navigation night basemap style
+        val map = ArcGISMap(BasemapStyle.ArcGISTopographic)
+        mapView.map = map
+
+        // locate the .stylx file in the device
+        val styleFile = File(provisionPath, getString(R.string.mil2525d_stylx))
+        // instantiate the dictionarySymbolStyle using the file path
+        val dictionarySymbolStyle = DictionarySymbolStyle.createFromFile(styleFile.absolutePath)
+
+        // locate the .geodatabase file in the device
+        val geodatabaseFile = File(provisionPath, getString(R.string.militaryoverlay_geodatabase))
+        // instantiate the geodatabase with the file path
+        val geodatabase = Geodatabase(geodatabaseFile.path)
+
+        lifecycleScope.launch {
+            // load the dictionary symbol style
+            dictionarySymbolStyle.load().getOrElse {
+                return@launch showError("Error loading DictionarySymbolStyle: ${it.message}")
+            }
+        }
+
+        lifecycleScope.launch {
+            // load the geodatabase
+            geodatabase.load().getOrElse {
+                showError("Error loading Geodatabase: ${it.message}")
+            }
```

Comment:
```suggestion
        lifecycleScope.launch {
            // load the dictionary symbol style
            dictionarySymbolStyle.load().getOrElse {
                return@launch showError("Error loading DictionarySymbolStyle: ${it.message}")
            }

            // load the geodatabase
            geodatabase.load().getOrElse {
                showError("Error loading Geodatabase: ${it.message}")
            }
```

Loading both `dictionarySymbolStyle` and the `geodatabase` in two different scopes would create a possibility that one would load before the other causing a race condition. Suggest just loading both in one coroutine scope.

Reply:
good catch, applied the suggestion